module_generator/README.md

```diff
@@ -0,0 +1,220 @@
+# Module Generator Template
+
+This is the Python module template for [CLI module generator](https://docs.viam.com/cli/#module).
+
+## How to Use
+
+1. Install cookiecutter: `pipx install cookiecutter`. If pipx is not an option, use `python -m pip install --user cookiecutter`.
+2. Call the template: `python3 module_generator/cookiecutter-template`.
+3. Fill out the prompts given (see below).
+4. The necessary files for the created module will be generated in a new directory `module_generator/<MODULE_NAME>`.
+
+NOTE: The imports in `module_generator/<MODULE_NAME>/src/__main__.py` may have to be adjusted. More details below in the example.
+
+## Prompts
+
+To create a module, the generator needs the user to fill out information about the module. The prompts can be seen in `cookiecutter-template/cookiecutter.json`. The public prompts that a user directly affects are as follows:
```

Comment:
Can we use the `__prompts__` configuration field to show the descriptive text for each prompt listed here? https://cookiecutter.readthedocs.io/en/stable/advanced/human_readable_prompts.html#human-readable-prompts

Reply:
Good suggestion! I almost missed that, although I'm not sure how many people will really see this since it's going to be run in the CLI. Doesn't hurt to add though :)

Reply:
Is the CLI going to prompt with a different interface and pass those arguments to this template generator?

Reply:
From what I know so far, we're not sure how we're going to tie things together yet. We'll be working through it and having meetings about it in the coming month.

Reply:
some details are TBD, but the expectation is that the CLI will wrap this `cookiecutter` code by creating an execuable via `pyinstaller`. I'd be pretty surprised if we moved away from that at a high level. So, the prompts here will be seen in the CLI.